src/http/event-http.js

```diff
@@ -1,9 +1,30 @@
+import _ from 'lodash';
 import * as eventCore from '../core/event-core';
 import {createJsonRoute} from '../util/express';
 import {assert} from '../validation';
 
 const getEvents = createJsonRoute(function(req, res) {
-  return eventCore.getEvents(assert(req.query, 'eventsParams'));
+  const eventParams = assert({
+    id: req.params.id,
+    city: req.query.city,
+  }, 'eventsParams');
+
+  const coreParams = _.merge(eventParams, {
+    client: req.client,
+  });
+
+  return eventCore.getEvents(coreParams).then(results => {
+    if (req.params.id !== undefined) {
+      if (results.length > 1) {
+        throw new Error('Unexpected number of rows');
+      }
```

Comment:
Or actually I guess not since it's a query param.. I guess the proper way would be to create an `:id` route for this